src/ArrayValidator.java

```diff
@@ -0,0 +1,9 @@
+public class ArrayValidator {
+    public static void validateSize(String[][] arr) throws ArraySizeException {
+        int expectedLength = 4;
+
+        if (arr.length != expectedLength) {
+            throw new ArraySizeException("Array size does not match expected size");
+        }
+    }
+}
```

Comment:
this logic can be placed in the doCalc method. No reason to have a method that throws exceptions only